lib/components/particles/accelerated_particle.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+
+import '../mixins/single_child_particle.dart';
+import '../particle_component.dart';
+import 'curved_particle.dart';
+
+/// A particle serves as a container for basic
+/// acceleration physics.
+/// [Offset] unit is logical px per second.
+/// speed = Offset(0, 100) is 100 logical pixels per second, down
+/// acceleration = Offset(-40, 0) will accelerate to left at rate of 40 px/s
+class AcceleratedParticle extends CurvedParticle with SingleChildParticle {
+  @override
+  Particle child;
+
+  final Offset acceleration;
+  Offset speed;
+  Offset position;
+
+  AcceleratedParticle({
+    @required this.child,
+    this.acceleration = Offset.zero,
+    this.speed = Offset.zero,
+    this.position = Offset.zero,
+    double lifespan,
+    Duration duration,
+  }) : super(lifespan: lifespan, duration: duration);
+
+  @override
+  void render(Canvas canvas) {
+    canvas.save();
+    canvas.translate(position.dx, position.dy);
+    super.render(canvas);
+    canvas.restore();
+  }
+
+  @override
+  void update(double t) {
+    speed += acceleration * t;
+    position += speed * t;
```

Comment:
I think you actually need to use another formula here, this is an approximation for very small 't' but for real life ts might look unrealistic. See here for the correct formula:
https://github.com/luanpotter/dino/blob/master/game/lib/main.dart#L38
basically speed is changed the same but position must be corrected using the derivative, the resulting effect is `position += speed * t - a t^2 / 2`
notice how your formula does not give the same results if called once with 10 or twice with 5. since we cannot control how small t is in real life, this might be noticiable

Reply:
Fixed, thank you!